### ISSUE TITLE
Unable to launch Firefox session on Windows

### ISSUE THREAD

**Issue by Vaelek** [httptoolkit/feedback#23]
Worked great for several days. Suddenly the "Fresh Firefox" option is labeled not available, and when I click it, Debug tools state that it is not installed. It most certainly is.

```
vendors~main.bundle.js:70 Error: Server error: Unexpected error value: "firefox is not installed in your system." at activateInterceptor
    at f (app.js:1)
```

Same results on 2 machines.
Windows 10 Pro, Firefox developer edition

---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
Interesting! That's very odd. The Windows browser detection is powered by https://github.com/vweevers/win-detect-browsers (via https://github.com/james-proxy/james-browser-launcher), but I haven't seen it fail like this before.

You say you're seeing this on two different machines? Did they both stop detecting it at the same time? When exactly?

There is a cached browser config file that's stored at `C:\Users\<Username>\AppData\Local\httptoolkit\Config\browsers.json`. Can you post the contents of that file here? Shouldn't be very sensitive, it's just the list of installed browsers & their paths, but it might give some clues. It's possible here that it has somehow cached an invalid result, although in theory that would've fixed itself.

Once you've grabbed the contents, try closing HTTP Toolkit, deleting that file, and then starting it again, and let me know if that resolves your issue.

**Comment by Vaelek** [comment]
It was working fine on Friday or Saturday, and no longer yesterday. I hadn't used it on the 2nd machine for a few days but when I went to try that figuring it had to be something on my main machine, I ran into the same issue. 

My browsers.json contains only IE

```json
{
  "browsers": [
    {
      "type": "ie",
      "name": "ie",
      "command": "C:\\Program Files (x86)\\Internet Explorer\\iexplore.exe",
      "version": "11.0.17763.1",
      "profile": false
    },
    {
      "type": "ie",
      "name": "ie",
      "command": "C:\\Program Files\\Internet Explorer\\iexplore.exe",
      "version": "11.0.17763.1",
      "profile": false
    }
  ]
}
```
My 2nd machine has the same with the addition of Chrome.

After removing the file and relaunching the toolkit, Opera is added (I installed Vivaldi and Opera hoping for non-system proxy settings to work around it) but Firefox is still missing. Same results on both machines.

However, given this information, I at least was able to manually add a section for Firefox and get it working again. 

I added
```json
    {
	  "type": "firefox",
	  "name": "firefox",
	  "command": "C:\\Program Files\\Mozilla Firefox\\firefox.exe",
	  "version": "66.0.3",
	  "profile": "C:\\Users\\jim\\AppData\\Local\\httptoolkit\\Config\\firefox-profile"
    }
```

**Comment by pimterry** [comment]
A quick update on this: I've been working on it, and Firefox should again be correctly detected on Windows by [@httptoolkit/browser-launcher](https://github.com/httptoolkit/browser-launcher). The only remaining issue is ensuring that native modules work correctly in the built server, which is blocked by https://github.com/vweevers/win-version-info/pull/7.

Once that's merged, win-version-info should work in the server, and I'll be able to release that to fix this completely.

**Comment by pimterry** [comment]
Now fixed :smiley: I've just released the updated server as 0.1.15 - if you open the app and wait a few minutes it will update in the background, and on the next run this will be resolved. You can check the current version by hovering over the 'HTTP Toolkit' logo in the sidebar.

Firefox not being detected will probably still be cached initially, but if you click the Firefox button anyway it'll forcibly refresh it, and from then on everything should work as normal.

Thanks for reporting this, hopefully this should all be sorted now. Let me know if you have any other trouble!
